src/analysis/inner_product_space/basic.lean

```diff
@@ -2322,3 +2323,53 @@ lemma submodule.orthogonal_family_self :
 | ff ff := absurd rfl
 
 end orthogonal
+
+namespace uniform_space.completion
+
+open uniform_space function
+
+instance {𝕜' E' : Type*} [topological_space 𝕜'] [uniform_space E'] [has_inner 𝕜' E'] :
+  has_inner 𝕜' (completion E') :=
+{ inner := curry $ (dense_inducing_coe.prod dense_inducing_coe).extend (uncurry inner) }
+
+@[simp] lemma coe_inner (a b : E) :
+  inner (a : completion E) (b : completion E) = (inner a b : 𝕜) :=
+(dense_inducing_coe.prod dense_inducing_coe).extend_eq
+  (continuous_inner : continuous (uncurry inner : E × E → 𝕜)) (a, b)
+
+protected lemma continuous_inner :
+  continuous (uncurry inner : completion E × completion E → 𝕜) :=
+begin
+  let inner' : E →+ E →+ 𝕜 :=
+  { to_fun := λ x, (innerₛₗ x).to_add_monoid_hom,
+    map_zero' := by ext x; exact inner_zero_left,
+    map_add' := λ x y, by ext z; exact inner_add_left },
+  have : continuous (λ p : E × E, inner' p.1 p.2) := continuous_inner,
+  rw [completion.has_inner, uncurry_curry _],
+  change continuous (((dense_inducing_to_compl E).prod (dense_inducing_to_compl E)).extend
+    (λ p : E × E, inner' p.1 p.2)),
+  exact (dense_inducing_to_compl E).extend_Z_bilin (dense_inducing_to_compl E) this,
+end
+
+protected lemma continuous.inner {α : Type*} [topological_space α]
+  {f g : α → completion E} (hf : continuous f) (hg : continuous g) :
+  continuous (λ x : α, inner (f x) (g x) : α → 𝕜) :=
+uniform_space.completion.continuous_inner.comp (hf.prod_mk hg : _)
+
+local attribute [continuity] uniform_space.completion.continuous.inner
+
+instance : inner_product_space 𝕜 (completion E) :=
```

Comment:
All the `continuity` calls here make this declaration pretty slow to compile. Replacing them using `continuity?` makes it fast, but obviously it's not as clean. Up to you or other reviewers which to prefer here.

Reply:
I vote for the faster version; I think timeouts are a bigger maintainability issue right now than slightly less readable code.

Reply:
I vote for the faster version; I think timeouts are a bigger maintainability issue right now than slightly less readable code.

Reply:
Oh sure, I just didn't know about `continuity?`. Let me change that